File: lib/configs/default_values.dart
// lib/constants/default_values.dart

/// ค่าเริ่มต้นที่ใช้ในแอป
class DefaultValues {
  // ===== Asset Defaults =====
  static const String description = '(ไม่มีคำอธิบาย)';
  static const String unknown = 'unknown';
  static const String environment = 'unknown';
  static const String mobility = 'unknown';
  static const String status = 'unknown';
  static const String owner = '';
  static const String costCenter = '';
  static const String costCenterName = '';
  static const String location = '';
  static const String imageUrl = '';
  static const String condition = 'ใช้งานปกติ (Normal)';
  static const String updatedBy = '';
  static const int currentStatus = 0;
  
  // ===== Audit Defaults =====
  static String get auditYear => DateTime.now().year.toString();
  static const String auditor = 'unknown';
  
  // ===== Temp Photo Defaults =====
  static const String tempDescription = 'Temp Photo';
  static const String tempLocation = 'Unknown Location';
}

File: AssetApp-Android/app/src/main/java/com/plbg/assetapp/domain/model/DefaultValues.kt
package com.plbg.assetapp.domain.model

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

object DefaultValues {
    const val DESCRIPTION = "(No description)"
    const val UNKNOWN = "unknown"
    const val ENVIRONMENT = "unknown"
    const val MOBILITY = "unknown"
    const val STATUS = "unknown"
    const val OWNER = ""
    const val COST_CENTER = ""
    const val COST_CENTER_NAME = ""
    const val LOCATION = ""
    const val IMAGE_URL = ""
    const val CONDITION = "Normal"
    const val UPDATED_BY = ""
    const val CURRENT_STATUS = 0
    val AUDIT_YEAR: String get() = LocalDateTime.now().year.toString()
}

fun LocalDateTime.toIsoString(): String =
    this.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME)

fun String.toLocalDateTimeSafe(): LocalDateTime? =
    runCatching { LocalDateTime.parse(this) }.getOrNull()